sqlglot/dialects/snowflake.py

```diff
@@ -1427,3 +1426,36 @@ def arrayagg_sql(self, expression: exp.ArrayAgg) -> str:
                 expr_sql = self.sql(exp.WithinGroup(this=expr_sql, expression=order))
 
             return expr_sql
+
+        def array_sql(self, expression: exp.Array) -> str:
+            expressions = expression.expressions
+
+            first_expr = seq_get(expressions, 0)
+            if isinstance(first_expr, exp.Select):
+                # SELECT AS STRUCT foo AS alias_foo -> ARRAY_AGG(OBJECT_CONSTRUCT('alias_foo', foo))
+                kind = first_expr.args.get("kind")
+                if kind and kind.upper() == "STRUCT":
+                    object_construct_args = []
+                    for expr in first_expr.expressions:
+                        if isinstance(expr, exp.Alias):
+                            # SELECT AS STRUCT foo AS alias_foo -> OBJECT_CONSTRUCT('alias_foo', foo)
+                            name = expr.this.name
+                            alias = f"'{expr.alias}'"
```

Comment:
Let's construct a proper `Literal.string(...)` here instead of hardcoding the string?